### DISCUSSION TITLE
[Security]Using 2 authentications modes (DB and LDAP)

### DISCUSSION THREAD

**Discussion by helicos441** [symfony/symfony#45695] [Q&A]
Hello all,

I am using Symfony 5.4 on a project for my company.
I use actually an entity authentication mode with form login for authenticate users. 

Next release, I need to use both LDAP authentication for company's users and entity authentication for other users for same login form. 
- Company's users use LDAP authentication and these password are not store on DB.
- Other users are using entity authentication and these password are store on DB.
- A flag on user table indicate if ldap authentication is used for an user or not.
- User data (first name, email, ...) are already load from database. LDAP is only used for authenticate user.

I found ldap authentication component and configure it but i don't find how to combine 2 authentication modes.
I don't find any solutions after search on web and Symfony documentation.

Thanks for all !

helicos

**Comment by mpeltier** [comment]
Hello,
The following thread may help you:
https://stackoverflow.com/questions/62898066/symfony-5-1-ldap-authentication-with-entity-user-provider
Mathieu
